README.md
# Forge UI

## Component library for [Forge](https://github.com/gamma-guys/forge) by [Gamma Guys Studio](https://github.com/gamma-guys)


CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file. See [standard-version](https://github.com/conventional-changelog/standard-version) for commit guidelines.

### 0.0.2 (2021-04-18)


### Features

* **stitches.config.ts && components:** configure and export proper stitches / dark theme configs ([1952449](https://github.com/gamma-guys/forge-ui/commit/1952449104300be257ae6f1dc44cf82fff3d1821))
* **stitchesconfig.ts:** create stitches config with theming, utils, breakpoints, etc ([fc63df1](https://github.com/gamma-guys/forge-ui/commit/fc63df17f6ab0656a97c1dd1df74e82844e6da01))


### Bug Fixes

* **rollup.config.js:** fix module imports throwing type error when `npm run build` is run ([0238f55](https://github.com/gamma-guys/forge-ui/commit/0238f552f5c6da446909c18f3fa88d4fcc61f49f))


src/stitchesConfig.ts
import { createCss } from "@stitches/react";

export const { theme, styled, css } = createCss({
  utils: {
    // @ts-ignore
    m: (config) => (value) => ({
      marginTop: value,
      marginRight: value,
      marginBottom: value,
      marginLeft: value,
    }),
    // @ts-ignore
    mt: (config) => (value) => ({
      marginTop: value,
    }),
    // @ts-ignore
    mr: (config) => (value) => ({
      marginRight: value,
    }),
    // @ts-ignore
    mb: (config) => (value) => ({
      marginBottom: value,
    }),
    // @ts-ignore
    ml: (config) => (value) => ({
      marginLeft: value,
    }),
    // @ts-ignore
    mx: (config) => (value) => ({
      marginLeft: value,
      marginRight: value,
    }),
    // @ts-ignore
    my: (config) => (value) => ({
      marginTop: value,
      marginBottom: value,
    }),
    // @ts-ignore
    br: (config) => (value) => ({
      borderRadius: value,
    }),
    // @ts-ignore
    size: (config) => (value) => ({
      width: value,
      height: value,
    }),
  },
  media: {
    xs: "(max-width: 500px)",
    sm: "(max-width: 768px)",
    md: "(max-width: 1024px)",
    lg: "(min-width: 1025px)",
  },
  theme: {
    colors: {
      bg: "#FFF9F6",
      font: "#17080d",

      darkBlueBase: "#242f3d",
      darkBlue50: "#ecf2fb",
      darkBlue100: "#cfd8e3",
      darkBlue200: "#b1bdcd",
      darkBlue300: "#92a3b9",
      darkBlue400: "#7389a5",
      darkBlue500: "#596f8b",
      darkBlue600: "#45566d",
      darkBlue700: "#313e4e",
      darkBlue800: "#1c2530",
      darkBlue900: "#050c15",

      orangeBase: "#f79465",
      orange50: "#ffece0",
      orange100: "#fecbb3",
      orange200: "#faab85",
      orange300: "#f68955",
      orange400: "#f36826",
      orange500: "#d94f0d",
      orange600: "#aa3d09",
      orange700: "#7a2a05",
      orange800: "#4b1900",
      orange900: "#1e0600",

      lightGreyBase: "#999999",
      lightGrey50: "#feeff2",
      lightGrey100: "#ded7d9",
      lightGrey200: "#c2bebf",
      lightGrey300: "#a7a5a6",
      lightGrey400: "#8c8c8c",
      lightGrey500: "#737373",
      lightGrey600: "#5a5959",
      lightGrey700: "#423f40",
      lightGrey800: "#2a2526",
      lightGrey900: "#17080d",
    },
  },
});

export const darkTheme = theme({
  colors: {
    bg: "#17080d",
    font: "#FFF9F6",
  },
});


src/stitches.config.ts
import { createCss } from "@stitches/react";

export const { theme, styled, css, global, getCssString } = createCss({
  utils: {
    // @ts-ignore
    m: (config) => (value) => ({
      marginTop: value,
      marginRight: value,
      marginBottom: value,
      marginLeft: value,
    }),
    // @ts-ignore
    mt: (config) => (value) => ({
      marginTop: value,
    }),
    // @ts-ignore
    mr: (config) => (value) => ({
      marginRight: value,
    }),
    // @ts-ignore
    mb: (config) => (value) => ({
      marginBottom: value,
    }),
    // @ts-ignore
    ml: (config) => (value) => ({
      marginLeft: value,
    }),
    // @ts-ignore
    mx: (config) => (value) => ({
      marginLeft: value,
      marginRight: value,
    }),
    // @ts-ignore
    my: (config) => (value) => ({
      marginTop: value,
      marginBottom: value,
    }),
    // @ts-ignore
    br: (config) => (value) => ({
      borderRadius: value,
    }),
    // @ts-ignore
    size: (config) => (value) => ({
      width: value,
      height: value,
    }),
  },
  media: {
    xs: "(max-width: 500px)",
    sm: "(max-width: 768px)",
    md: "(max-width: 1024px)",
    lg: "(min-width: 1025px)",
    dark: "(prefers-color-scheme: dark)",
  },
  theme: {
    radii: { round: "50%" },
    colors: {
      bg: "#FFF9F6",
      font: "#17080d",

      darkBlueBase: "#242f3d",
      darkBlue50: "#ecf2fb",
      darkBlue100: "#cfd8e3",
      darkBlue200: "#b1bdcd",
      darkBlue300: "#92a3b9",
      darkBlue400: "#7389a5",
      darkBlue500: "#596f8b",
      darkBlue600: "#45566d",
      darkBlue700: "#313e4e",
      darkBlue800: "#1c2530",
      darkBlue900: "#050c15",

      orangeBase: "#f79465",
      orange50: "#ffece0",
      orange100: "#fecbb3",
      orange200: "#faab85",
      orange300: "#f68955",
      orange400: "#f36826",
      orange500: "#d94f0d",
      orange600: "#aa3d09",
      orange700: "#7a2a05",
      orange800: "#4b1900",
      orange900: "#1e0600",

      lightGreyBase: "#999999",
      lightGrey50: "#feeff2",
      lightGrey100: "#ded7d9",
      lightGrey200: "#c2bebf",
      lightGrey300: "#a7a5a6",
      lightGrey400: "#8c8c8c",
      lightGrey500: "#737373",
      lightGrey600: "#5a5959",
      lightGrey700: "#423f40",
      lightGrey800: "#2a2526",
      lightGrey900: "#17080d",
    },
  },
});

export const darkTheme = theme({
  colors: {
    bg: "#17080d",
    font: "#FFF9F6",
  },
});

export const globalStyles = global({
  "@dark": {
    // notice the `media` definition on the stitches.config.ts file
    ":root:not(.light)": {
      // @ts-ignore
      ...Object.keys(darkTheme.colors).reduce((varSet, currentColorKey) => {
        // @ts-ignore
        const currentColor = darkTheme.colors[currentColorKey];
        const currentColorValue =
          currentColor.value.substring(0, 1) === "$"
            ? `$colors${currentColor.value}`
            : currentColor.value;

        return {
          [currentColor.variable]: currentColorValue,
          ...varSet,
        };
      }, {}),
    },
  },
});
